lib/streamlit/auth_util.py

```diff
@@ -172,6 +172,12 @@ def validate_auth_credentials(provider: str) -> None:
 
     provider_section = auth_section.get(provider)
 
+    if "_" in provider:
+        raise StreamlitAuthError(
```

Comment:
In case this is more of a temporary safeguard, maybe add a todo comment here to allow this usage.

Reply:
Done!